### PR TITLE
fix(hypergraphs): complete bounded minimal-transversal enumeration

### LINKED ISSUE #3594: [Hypergraphs] Enumerate bounded-rank minimal transversals
## Observed demand

One retained query asks for the complete enumeration of all inclusion-minimal transversals of size at most three. Current `hypergraph.minimum_transversal.compute` returns one minimum-cardinality hitting set. It cannot recover other inclusion-minimal transversals, including larger ones, and repeatedly excluding solutions through a solver would move completeness and deduplication to the caller.

## Proposed operation

```text
hypergraph.minimal_transversals.bounded_rank.enumerate
```

Input a canonical finite hypergraph and a maximum returned cardinality `k`. Return every inclusion-minimal transversal `T` with `|T| <= k`, in canonical vertex order, together with the complete cardinality profile for that slice. Each row must:

- intersect every source hyperedge; and
- cease to be a transversal after deleting any member.

The operation claims completeness only for minimal transversals through `k`; it does not claim there are no larger minimal transversals. The result composes with set-family orbit, optimization, and dual-hypergraph operations. `hypergraph.minimum_transversal.compute` remains the cheaper scalar/witness operation.

## Bounds and fixtures

Preflight `sum_(i=0)^k binomial(n,i)` candidate subsets or a sound bound for the chosen transversal-generation algorithm, incidence checks, predicted result rows/incidences, label bytes, and total output. A cheap minimality rejection may prune work; accepted negative slices still require complete search.

Test a single edge, disjoint edges, nested edges, a triangle graph viewed as a hypergraph, duplicate-free canonical order, and a case with one minimal transversal below `k` and another above it. Check every row directly and compare small cases with brute-force subset enumeration. Operational exhaustion cannot become an empty complete slice.

All unrestricted hypergraph dualization, transversal counting without materialization, and minimum-weight hitting sets remain separate contracts.

### PR DESCRIPTION
**Pull request by morluto**
## Summary

Closes #3594.

- Remove the JSON/encoded-result byte admission from bounded minimal-transversal enumeration; mathematical admission now uses candidate cardinality, candidate-edge/minimality work, Sperner/LYM antichain row bounds, and output incidence bounds.
- Preserve exact degenerate behavior for edge-free and empty-edge families, including a complete requested-rank cardinality profile.
- Bind returned rows to the source vertex axis and profile, reject malformed direct native requests, and add cooperative owner deadlines/checkpoints.
- Keep catalog/API composition on the typed request/result contract.

## Validation

- `uv run pytest -q tests/math/combinatorics/finite_structures/hypergraphs/test_minimal_transversal_enumeration.py` (11 passed)
- `uv run pytest -q tests/math/combinatorics/finite_structures/hypergraphs tests/catalog/test_catalog.py tests/catalog/test_tool_manifests.py tests/dispatch/test_dispatch.py` (319 passed)
- `uv run pytest -q tests/tooling/test_architecture.py tests/tooling/test_affected_validation.py` (51 passed)
- `make affected AFFECTED_BASE=origin/main` (ruff check/format, mypy, and 281 hypergraph tests passed)

Head SHA: `724abfa2f2bd7f8a620b376c76061f9202d964f9`

### CODE REVIEW

File: tests/math/combinatorics/finite_structures/hypergraphs/test_minimal_transversal_enumeration.py [hunk 1]
```diff
@@ -34,3 +71,111 @@ def test_empty_edge_has_no_transversal() -> None:
...
+    )
+    with pytest.raises(OperationResourceAdmissionError, match="result rows"):
+        enumerate_minimal_transversals(
+            MinimalTransversalEnumerationRequest(
+                hypergraph=source, maximum_cardinality=8
+            )
+        )
+
+
+def test_candidate_edge_and_minimality_work_is_admitted_before_search() -> None:
```
[the remark this answers is not in the document]
Comment (morluto) [reply tests/math/combinatorics/finite_structures/hypergraphs/test_minimal_transversal_enumeration.py:263]: Fixed in 1491999de. The regression now isolates minimality accounting: candidate-edge work alone is below the cap while combined candidate and minimality work is rejected before enumeration. The accepted parity case remains in the focused suite.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 2]
```diff
@@ -1,67 +1,308 @@
...
+    total_work = candidate_edge_work + minimality_work
+
+    # Minimal transversals form an antichain. By the LYM inequality, the
+    # number of rows in ranks 1..k is at most the largest binomial coefficient
+    # among those ranks.
+    possible_rows = max(
+        (comb(len(vertices), size) for size in range(1, maximum + 1)),
+        default=0,
+    )
+    if possible_rows > MAX_ENUMERATED_TRANSVERSALS:
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Fixed in 1491999de. A bounded closed single-edge path returns the complete source-axis ordered singleton family, preserving the inexpensive valid slice rather than applying the global Sperner envelope. Focused and full hypergraph tests pass.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 3]
```diff
@@ -1,67 +1,308 @@
...
+    profile = tuple(
+        MinimalTransversalCardinalityCount(cardinality=cardinality, count=count)
+        for cardinality, count in enumerate(counts)
+    )
     return MinimalTransversalEnumerationResult(
         hypergraph=request.hypergraph,
         maximum_cardinality=request.maximum_cardinality,
-        transversals=tuple(results),
+        transversals=results,
+        cardinality_profile=profile,
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Fixed in 1491999de. Native execution now checkpoints after structural result construction, and the deadline regression proves an expired request cannot return after validation work. Focused and full hypergraph tests pass.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 4]
```diff
@@ -1,67 +1,373 @@
...
+        results: tuple[tuple[str, ...], ...] = ((),)
+    elif has_empty_edge or maximum == 0 or len(forced) > maximum:
+        results = ()
+    elif not remaining_edges:
+        results = (forced_row,)
+    elif len(remaining_edges) == 1:
+        results = tuple(
+            tuple(vertex for vertex in vertices if vertex in forced or vertex == extra)
+            for extra in vertices
+            if extra in remaining_edges[0]
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py:458]: Addressed by the branch's earlier commits (84f44d9ce, 7bdc2b835): `enumerate_minimal_transversals` now tests `len(forced) >= maximum` before the single-remaining-edge shortcut, so edges `{a}` and `{b,c}` with `maximum_cardinality=1` return the complete empty slice. Verified directly: the request returns `transversals=()` with an empty cardinality profile.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 5]
```diff
@@ -1,67 +1,436 @@
...
+    if len(forced) >= maximum:
+        return maximum, remaining_unique, forced, False, False
+    # Cardinality-1 search never needs domination. Equal-cardinality distinct
+    # remaining edges are already an antichain, so skip the pairwise subset scan.
+    edge_sizes = {len(edge) for edge in remaining_unique}
+    need_domination = maximum > 1 and len(edge_sizes) > 1
+    domination_work = (
+        _domination_comparison_count(remaining_unique) if need_domination else 0
+    )
+    edges = _minimal_edges(remaining_unique) if need_domination else remaining_unique
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py:354]: Fixed in 2f380ac65. `_admit_enumeration` now computes a combined domination-plus-search bound from the pre-domination edge set and checks it before `_minimal_edges` runs, so a mixed-size antichain that would exceed the shared work bound is refused without executing the subset scan. The post-domination tight bound is retained.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 6]
```diff
@@ -1,67 +1,436 @@
...
+            location=("maximum_cardinality",),
+            code="hypergraph.minimal_transversal.work_bound",
+            message=(
+                "candidate-edge membership and minimality work has "
+                f"{total_work} checks; maximum is "
+                f"{MAX_TRANSVERSAL_ENUMERATION_WORK}"
+            ),
+        )
+
+    output_incidences = possible_rows * maximum
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Fixed in 2f380ac65. `output_incidences` now uses `min(maximum, len(forced) + len(free_vertices))` instead of the source-wide `maximum`, so isolated vertices removed from `free_vertices` are not charged. The finding's 256-vertex / two 8-edge case returns its 64 two-vertex rows instead of being rejected.

File: tests/math/combinatorics/finite_structures/hypergraphs/test_minimal_transversal_enumeration.py [hunk 7]
```diff
@@ -34,3 +81,577 @@ def test_empty_edge_has_no_transversal() -> None:
...
+    candidate_count = sum(
+        comb(vertex_count, size) for size in range(1, maximum_cardinality + 1)
+    )
+    weighted_candidate_count = sum(
+        size * comb(vertex_count, size) for size in range(1, maximum_cardinality + 1)
+    )
+    charged = {
+        "candidate_edge": candidate_count * len(source.edges),
+        "minimality": weighted_candidate_count * len(source.edges),
+    }
```
[the remark this answers is not in the document]
Comment (morluto) [reply tests/math/combinatorics/finite_structures/hypergraphs/test_minimal_transversal_enumeration.py:334]: Fixed in 2f380ac65. Added `test_accepted_mixed_size_request_charges_domination_parity`, which instruments the size-ordered domination comparison count and asserts an accepted mixed 2-set/3-set request's executed domination comparisons are within its admission charge.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 8]
```diff
@@ -1,67 +1,373 @@
...
+def _admit_enumeration(
+    request: MinimalTransversalEnumerationRequest,
+) -> tuple[int, tuple[frozenset[str], ...], frozenset[str], bool, bool]:
+    """Admit exhaustive work after edge dedup and singleton presolve."""
+
     vertices = request.hypergraph.vertices
     maximum = min(request.maximum_cardinality, len(vertices))
-    candidate_bound = sum(comb(len(vertices), size) for size in range(maximum + 1))
-    if candidate_bound > MAX_TRANSVERSAL_ENUMERATION_CANDIDATES:
+    edges = _minimal_edges(_unique_edges(request))
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Addressed by the branch's earlier commits (84f44d9ce, 7bdc2b835): `enumerate_minimal_transversals` now tests `len(forced) >= maximum` before the single-remaining-edge shortcut, so edges `{a}` and `{b,c}` with `maximum_cardinality=1` return the complete empty slice. Verified directly: the request returns `transversals=()` with an empty cardinality profile.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 9]
```diff
@@ -1,67 +1,409 @@
...
+    """Admit exhaustive work after edge dedup and singleton presolve."""
+
     vertices = request.hypergraph.vertices
     maximum = min(request.maximum_cardinality, len(vertices))
-    candidate_bound = sum(comb(len(vertices), size) for size in range(maximum + 1))
-    if candidate_bound > MAX_TRANSVERSAL_ENUMERATION_CANDIDATES:
+    unique_edges = _unique_edges(request)
+    # Cardinality-1 search never needs domination: every hitting singleton is
+    # already minimal, and the residual scan is linear in incidences.
+    edges = unique_edges if maximum <= 1 else _minimal_edges(unique_edges)
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Addressed by the branch's earlier commit 7bdc2b835 ("exhaust forced rank before domination") and the tightened single-edge ordering; verified the `{a}`/`{b,c}`, `maximum_cardinality=1` case returns the complete empty slice.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 10]
```diff
@@ -1,67 +1,409 @@
...
+
+
+def _minimal_edges(
+    edges: tuple[frozenset[str], ...],
+) -> tuple[frozenset[str], ...]:
+    """Drop dominated edges after admitting the pairwise subset scan."""
+
+    edge_count = len(edges)
+    comparison_count = edge_count * max(edge_count - 1, 0)
+    if comparison_count > MAX_TRANSVERSAL_ENUMERATION_WORK:
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Addressed by the branch's earlier commit 7bdc2b835 ("exhaust forced rank before domination") and the tightened single-edge ordering; verified the `{a}`/`{b,c}`, `maximum_cardinality=1` case returns the complete empty slice.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 11]
```diff
@@ -1,67 +1,515 @@
...
+            for size in range(0 if forced else 1, pre_free_maximum + 1)
+        )
+        pre_constraint_count = len(pre_edges) + len(forced)
+        pre_minimality_work = sum(
+            (len(forced) + size) * pre_constraint_count * comb(len(pre_free), size)
+            for size in range(0 if forced else 1, pre_free_maximum + 1)
+        )
+        if (
+            pre_candidate_count * len(pre_edges) + pre_minimality_work + domination_work
+            > MAX_TRANSVERSAL_ENUMERATION_WORK
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py]: Fixed in 15a89aea0 and covered by 1d6cd0d43. My earlier combined pre-admission bound was unsound: it charged the pre-domination edge set, so a family that domination reduces before the search was rejected. The pre-scan check now bounds only the domination subset comparisons (the sole unadmitted phase) against `MAX_TRANSVERSAL_ENUMERATION_WORK`; the combined candidate/minimality bound is enforced after `_minimal_edges` with the tighter minimal edge set. The rank-3 dominated-superset case returns `('a',), ('b',)` again. Regression: `test_dominated_supersets_reduce_before_charging_search`.

File: src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py [hunk 12]
```diff
@@ -1,67 +1,492 @@
...
+        raise OperationResourceAdmissionError(
+            location=("hypergraph", "edges"),
+            code="hypergraph.minimal_transversal.antichain_work_bound",
+            message=(
+                "edge-domination presolve has "
+                f"{domination_work} subset comparisons; maximum is "
+                f"{MAX_TRANSVERSAL_ENUMERATION_WORK}"
+            ),
+        )
+    edges = _minimal_edges(remaining_unique) if need_domination else remaining_unique
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/combinatorics/finite_structures/hypergraphs/_transversal_enumeration.py:354]: Partially addressed in 782ad3a55, with a soundness constraint worth stating. A *sound* combined pre-admission bound must upper-bound the post-domination search work; the pre-domination edge set makes it an upper bound, but that is exactly the unsound estimate the earlier thread `h3Iyv` reproduced as a false rejection (the rank-3 dominated-superset family reduced to `{a,b}` but was charged for all 7,071 removable edges). Charging only the domination scan pre-scan and the combined bound post-scan avoids that regression.

To bound the pre-scan cost in the case you cite, `_minimal_edges` now aborts the size-ordered scan as soon as its comparisons reach `MAX_TRANSVERSAL_ENUMERATION_WORK`, and it still checkpoints every 256 comparisons. The 3,500/3,500 fixture remains admitted work (12.25M comparisons, under the 50M limit) and is refused by the combined bound with the `total_work` message the existing test asserts. There is no cheap sound lower bound on the post-reduction search work that would let a single edge be rejected pre-scan without risking a false rejection.
